TestForwardTSOUnexpectedToFollower2 is unstable

## Flaky Test

### Which jobs are failing

```
--- FAIL: TestForwardTSOUnexpectedToFollower2 (8.36s)
    server_test.go:446: 
        	Error Trace:	/home/runner/work/pd/pd/tests/integrations/mcs/tso/server_test.go:446
        	            				/home/runner/work/pd/pd/tests/integrations/mcs/tso/server_test.go:485
        	            				/home/runner/work/pd/pd/tests/integrations/mcs/tso/server_test.go:440
        	Error:      	Received unexpected error:
        	            	rpc error: code = Unknown desc = [PD:tso:ErrGenerateTimestamp]generate timestamp failed, requested pd is not leader of cluster
        	Test:       	TestForwardTSOUnexpectedToFollower2
    server_test.go:313: 
        	Error Trace:	/home/runner/work/pd/pd/tests/integrations/mcs/tso/server_test.go:313
        	            				/opt/hostedtoolcache/go/1.21.12/x64/src/runtime/panic.go:541
        	            				/opt/hostedtoolcache/go/1.21.12/x64/src/testing/testing.go:999
        	            				/home/runner/work/pd/pd/tests/integrations/mcs/tso/server_test.go:446
        	            				/home/runner/work/pd/pd/tests/integrations/mcs/tso/server_test.go:485
        	            				/home/runner/work/pd/pd/tests/integrations/mcs/tso/server_test.go:440
        	Error:      	Should be empty, but was [{"service-addr":"http://127.0.0.1:36189/","version":"None","git-hash":"None","deploy-path":"/home/runner/work/pd/pd/tests/integrations/mcs/tso","start-timestamp":1721034672} {"service-addr":"http://127.0.0.1:46735/","version":"None","git-hash":"None","deploy-path":"/home/runner/work/pd/pd/tests/integrations/mcs/tso","start-timestamp":1721034673}]
        	Test:       	TestForwardTSOUnexpectedToFollower2
FAIL
```

### CI link

https://github.com/tikv/pd/actions/runs/9936654996/job/27445363927?pr=8396


### Reason for failure (if possible)


### Anything else

--- a/server/grpc_service.go
+++ b/server/grpc_service.go
@@ -57,7 +57,7 @@ import (
 
 const (
 	heartbeatSendTimeout          = 5 * time.Second
-	maxRetryTimesRequestTSOServer = 3
+	maxRetryTimesRequestTSOServer = 6
 	retryIntervalRequestTSOServer = 500 * time.Millisecond
 	getMinTSFromTSOServerTimeout  = 1 * time.Second
 	defaultGRPCDialTimeout        = 3 * time.Second

